pulsar-client/src/main/java/org/apache/pulsar/client/impl/PartitionedConsumerImpl.java

```diff
@@ -437,6 +438,20 @@ public void redeliverUnacknowledgedMessages(Set<MessageIdImpl> messageIds) {
         }
     }
 
+    @Override
+    public void seek(MessageId messageId) throws PulsarClientException {
+        try {
+            seekAsync(messageId).get();
+        } catch (ExecutionException | InterruptedException e) {
+            throw new PulsarClientException(e);
```

Comment:
`e.getCause()`?

Reply:
If it's interrupted we might not have the "cause". Though I can refactor and have 2 sections here.